core/src/main/java/feign/stream/StreamDecoder.java

```diff
@@ -35,35 +35,41 @@
  * <pre>
  * <code>
  * Feign.builder()
- *   .decoder(StreamDecoder.create(JacksonIteratorDecoder.create()))
+ *   .decoder(StreamDecoder.create(new Decoder.Default()))
  *   .doNotCloseAfterDecode() // Required for streaming
  *   .target(GitHub.class, "https://api.github.com");
  * interface GitHub {
  *  {@literal @}RequestLine("GET /repos/{owner}/{repo}/contributors")
  *   Stream<Contributor> contributors(@Param("owner") String owner, @Param("repo") String repo);
  * }</code>
  * </pre>
+ * 
+ * @author mroccyen
  */
 public final class StreamDecoder implements Decoder {
 
-  private final Decoder iteratorDecoder;
+  private final Decoder delegate;
+  private IteratorDecoder iteratorDecoder;
 
-  StreamDecoder(Decoder iteratorDecoder) {
-    this.iteratorDecoder = iteratorDecoder;
+  public StreamDecoder(Decoder delegate) {
+    this.delegate = delegate;
   }
 
   @Override
   public Object decode(Response response, Type type)
       throws IOException, FeignException {
-    if (!(type instanceof ParameterizedType)) {
-      throw new IllegalArgumentException("StreamDecoder supports only stream: unknown " + type);
+    if (!isStream(type)) {
+      return delegate.decode(response, type);
     }
-    ParameterizedType streamType = (ParameterizedType) type;
-    if (!Stream.class.equals(streamType.getRawType())) {
-      throw new IllegalArgumentException("StreamDecoder supports only stream: unknown " + type);
+    IteratorDecoder iteratorDecoder;
```

Comment:
Let's not do this.

If anyone want to use same decoder for both, they can pass both on the constructor.

Reply:
Can that provide two constructors? one uses same decoder for both, the other uses two decoder for delegate and iterator?

Reply:
Another scenario, according to the return value of the 'iteratorDecoder.decode' to determine whether it is a 'Stream' type, and then go through different processing.

Reply:
Sure thing, as long you keep the original behavior

Reply:
Thanks, I get it.